tests/accelerators/test_ipu.py

```diff
@@ -37,6 +37,11 @@
     import poptorch
 
 
+pytestmark = pytest.mark.filterwarnings(
+    r"ignore:Detected call of `lr_scheduler.step\(\)` before `optimizer.step\(\)`:UserWarning"
+)
+
+
```

Comment:
Haven't looked into this further, but the reason for the warning could be a similar reason to the one in ddp_sharded as commented.

```python
FAILED tests/accelerators/test_ipu.py::test_all_stages[1] - UserWarning: Dete...
FAILED tests/accelerators/test_ipu.py::test_all_stages[4] - UserWarning: Dete...
FAILED tests/accelerators/test_ipu.py::test_stages_correct - UserWarning: Det...
FAILED tests/accelerators/test_ipu.py::test_autoreport - UserWarning: Detecte...
FAILED tests/accelerators/test_ipu.py::test_manual_poptorch_dataloader - User...
FAILED tests/accelerators/test_ipu.py::test_manual_poptorch_opts - UserWarnin...
FAILED tests/accelerators/test_ipu.py::test_manual_poptorch_opts_custom - Use...
FAILED tests/accelerators/test_ipu.py::test_default_opts - UserWarning: Detec...
```
https://dev.azure.com/PytorchLightning/pytorch-lightning/_build/results?buildId=59459&view=logs&j=84dafd16-84ac-5ac7-3837-ae3525208541&t=dade7e55-b49c-56cd-3b84-da61e40b580f&l=12152